catroid/src/org/catrobat/catroid/ui/adapter/LookAdapter.java

```diff
@@ -59,7 +60,14 @@ public long getItemId(int position) {
 			return INVALID_ID;
 		}
 		LookData item = getItem(position);
-		return idMap.get(item);
+
+		for (Map.Entry<LookData, Integer> e : idMap.entrySet()) {
+			if (item.equals(e.getKey())) {
+				return e.getValue();
+			}
+		}
+
+		return INVALID_ID;
```

Comment:
It seems like the HashMap throws a NullPointerException, even if the element is in the map (so it is equal by the definition of LookData.equal() ). So I check the entries myself, and it seems to work.